core/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -896,6 +898,46 @@ public void testRecoverFromStore() throws IOException {
         closeShards(newShard);
     }
 
+    /* This test just verifies that we fill up local checkpoint up to max seen seqID on primary recovery */
+    public void testRecoverFromStoreWithNoOps() throws IOException {
+        final IndexShard shard = newStartedShard(true);
```

Comment:
I think we can introduce a variant of `indexDoc` called `indexDocOnReplica` which takes a seq# as a parameter. This will remove the need for the extra shard. wdyt?